### PR TITLE
Make expose use introspection to grab the port value if possible.

### PR DESCRIPTION
**Pull request by brendandburns**
Also improve service printing to include public IP addresses.

### CODE REVIEW

**Comment by bprashanth** [comment]
Jeff, because kubectl

File: pkg/kubectl/cmd/expose.go [hunk 1]
```diff
@@ -113,11 +110,23 @@ func RunExpose(f *Factory, out io.Writer, cmd *cobra.Command, args []string) err
...
+		}
+		if util.GetFlagInt(cmd, "port") < 0 {
+			ports, err := f.PortsForResource(mapping, namespace, name)
+			if err != nil {
+				return err
+			}
+			if len(ports) == 0 {
+				return util.UsageError(cmd, "couldn't find a suitable port via --port flag or introspection")
+			}
+			params["port"] = ports[0]
```
Comment (jlowdermilk) [inline pkg/kubectl/cmd/expose.go]: This means expose will introspect and select the port of the first container in the case of a multi-container pod. If that is the intended behavior, it should be documented in the help string, though tbh, I think you should fail here if there is more than one port, as we don't know what the user actually intends.

File: pkg/kubectl/resource_printer.go [hunk 2]
```diff
@@ -394,9 +394,21 @@ func printService(svc *api.Service, w io.Writer) error {
 		formatLabels(svc.Spec.Selector), svc.Spec.PortalIP, svc.Spec.Ports[0].Port, svc.Spec.Ports[0].Protocol); err != nil {
 		return err
 	}
-	for i := 1; i < len(svc.Spec.Ports); i++ {
+	count := len(svc.Spec.Ports)
+	if len(svc.Spec.PublicIPs)+1 > count {
```
Comment (jlowdermilk) [inline pkg/kubectl/resource_printer.go]: Uncomfortably easy to introduce an off-by-one error here. I followed through the logic and it looks correct, but you should add a unit test.

Comment (brendandburns) [inline pkg/kubectl/resource_printer.go]: Refactored the code to make it much, much cleaner.  Also added test.

File: pkg/kubectl/cmd/expose.go (hunk 1 above)
Comment (brendandburns) [inline pkg/kubectl/cmd/expose.go]: done.

**Comment by brendandburns** [comment]
Comments addressed.  ptal.

File: pkg/kubectl/resource_printer.go (hunk 2 above)
Comment (jlowdermilk) [inline pkg/kubectl/resource_printer.go]: Much better, thank you.

**Comment by jlowdermilk** [comment]
LGTM. Will merge on green.

**Pull request merged**